datacube_stats/main.py

```diff
@@ -239,8 +239,11 @@ def from_configuration_file(cls, config, index=None, tile_indexes=None, output_l
         :return: read to run StatsApp
         """
         input_region = config.get('input_region')
-        if tile_indexes and not input_region:
-            input_region = {'tiles': tile_indexes}
+        if tile_indexes:
```

Comment:
Notice that `config.get('input_region')` might return `None` since there
may be no input region specified. Both of the checks below will then crash.

So how about:
```python
if tile_indexes is not None:
    if input_region is None:
        input_region = {'tiles': tile_indexes'}
    elif 'geometry' in input_region:
        input_region['tiles'] = tile_indexes
```
```